sshd-common/src/main/java/org/apache/sshd/common/keyprovider/KeyPairProvider.java

```diff
@@ -122,7 +132,7 @@ default KeyPair loadKey(SessionContext session, String type)
     /**
      * @param session The {@link SessionContext} for invoking this load command - may
      * be {@code null} if not invoked within a session context (e.g., offline tool).
-     * @return The available {@link Iterable} key types in preferred order - never {@code null}
```

Comment:
Please note that **order** is assumed here so to remove this documentation and/or behavior is a major change - especially for KEX.

Reply:
This diff is very misleading. The line 125 was not changed, the change was in the doc of org.apache.sshd.common.keyprovider.KeyPairProvider#getKeyTypes in line 135, which returns a HashSet, which does NOT keep any order. The order is important in the other "part" later in AbstractServerSession: "expected" vs. "available" signatures...

Reply:
After reviewing your comment I  can live with the new phrasing